unity-renderer/Assets/Scripts/MainScripts/DCL/AvatarSystem/Visibility.cs

```diff
@@ -11,16 +11,18 @@ public class Visibility : IVisibility
 
         private Renderer combinedRenderer = null;
         private List<Renderer> facialFeatures = null;
+        private IBaseAvatar baseAvatar = null;
 
         /// <summary>
         /// Bind a set of renderers, previous renderers enabled wont be modified
         /// </summary>
         /// <param name="combinedRenderer"></param>
         /// <param name="facialFeatures"></param>
-        public void Bind(Renderer combinedRenderer, List<Renderer> facialFeatures)
+        public void Bind(Renderer combinedRenderer, List<Renderer> facialFeatures, IBaseAvatar baseAvatar = null)
```

Comment:
This creates an unneeded coupling between `Visibility` and `IBaseAvatar`.

I'd handle the visibility for the BaseAvatar in `AvatarWithHologram.AddVisibilityConstrain` and `AvatarWithHologram.RemoveVisibilityConstrain` and remove this dependency.

Reply:
Done! I had to add an `if` to check the key that is being addded since there are different ones that may affect the `AvatarWithHologram`. Maybe turning these consts strings into an enum somewhere would be nice?